cpp/src/unary/math_ops.cu

```diff
@@ -234,6 +234,21 @@ struct DeviceNot {
   }
 };
 
+// negation
+
+struct DeviceNegate {
+  template <typename T>
+  std::enable_if_t<std::is_signed_v<T>, T> __device__ operator()(T data)
+  {
+    return -data;
+  }
+  template <typename T>
+  std::enable_if_t<!std::is_signed_v<T>, T> __device__ operator()(T data)
+  {
+    return data;
+  }
```

Comment:
I think we can just not have this overload.

Reply:
Hmm template deduction failed without it